include/net/golioth/lightdb.h

```diff
@@ -158,6 +158,46 @@ int golioth_lightdb_delete_cb(struct golioth_client *client, const uint8_t *path
  */
 int golioth_lightdb_delete(struct golioth_client *client, const uint8_t *path);
 
+/**
+ * @brief Push value to Golioth's LightDB Stream (callback based)
+ *
+ * Asynchronously push new value to LightDB Stream and let @p cb be invoked when server
+ * acknowledges it or some error condition happens.
+ *
+ * @param client Client instance
+ * @param path LightDB Stream resource path
+ * @param format Format of payload
+ * @param data Payload data
+ * @param data_len Payload length
+ * @param[in] cb Callback executed on response received, timeout or error
+ * @param[in] user_data User data passed to @p cb
+ *
+ * @retval 0 On success
+ * @retval <0 On failure
+ */
+int golioth_stream_push_cb(struct golioth_client *client, const uint8_t *path,
```

Comment:
Nice decision to use the word "push", very descriptive.

Reply:
Was struggling with the name for this function. I decided to drop `lightdb` component from it. This is something that `goliothctl` does as well, with `goliothctl stream` subcommand. Additionally it makes API calls much smaller.

I guess we should move those functions to `stream.c` and `net/golioth/stream.h` files, right?

Reply:
Unfortunately we are not 100% consistent. IMO one of the problems causing that is that one service name "LightDB Stream" has another service "LightDB" in its name. That is confusing. People tend to use short names (abbreviations), so sometimes "LightDB" might be used when refering to "LightDB Stream". Docs are all good now, but I mean informal use of it (like between team members, during a call, etc.). I heard at least few times a question like "by LightDB you mean LightDB State, right?".

So in theory we have those 3 most commonly used terminology that is around:

1) LightDB | LightDB Stream

This is most common and it is what https://docs.golioth.io uses. IMO the biggest downside of that is the use of "LightDB" in both. In my engineer's mind, it would make sense if the latter (LightDB Stream) would be a subset of the former (LightDB), but not when it is a totally different service. The origins of both are similar, but I think that the use case of both is so different, that names of those services do not reflect that enough.

2) LightDB State | LightDB Stream

Not sure this is used formally, but it is used sometimes. The advantage is that using one of those names is always clear which service we are talking about.

3) LightDB | Stream

Yeah, we use that as well. Here are examples:
- `goliothctl lightdb` and `goliothctl stream` commands
- https://docs.golioth.io/reference/rest-api/openapi/ uses `.../{deviceId}/data` (LightDB) and `.../{deviceId}/stream` (LightDB Stream) endpoints.

I like it mostly because it is short, single word. All API calls are shorter that way. Also, there is no confusion when talking about "LightDB", as there is only one service with that word in name.

Reply:
We can stick with the names you are proposing. There's still not much consistency in docs and other places, but we won't be able to solve that problem right now. The names you've got make sense and are short.

> I guess we should move those functions to stream.c and net/golioth/stream.h files, right?

Yeah, I think I would prefer this, to make it easier to find them and to be consistent with the API names.

Reply:
Created `stream.h` and `stream.c` and left `golioth_stream_push{,_cb}` APIs, as originally proposed.